Editor/Overlays/CinemachineToolbarOverlay.cs

```diff
@@ -1,15 +1,213 @@
+        {
+            return new GUIContent()
+            {
+                image = AssetDatabase.LoadAssetAtPath<Texture2D>(ScriptableObjectUtility.CinemachineRealativeInstallPath
+                    + "/Editor/EditorResources/TrackedObjectOffset.png"),
+                tooltip = "Tracked Object Offset Tool",
+            };
+        }
+    }
+
+    /// <summary>
+    /// To add your custom tools (EditorToolbarElement) to the Cinemachine Tool Settings toolbar,
+    /// set CinemachineToolSettingsOverlay.customToolbarItems with your custom tools' IDs.
+    ///
+    /// By default, CinemachineToolSettingsOverlay.customToolbarItems is null.
+    /// </summary>
+    [Overlay(typeof(SceneView), "Cinemachine Tool Settings")]
+    [Icon("Packages/com.unity.cinemachine/Gizmos/cm_logo.png")]
+    public class CinemachineToolSettingsOverlay : Overlay, ICreateToolbar
+    {
+        static readonly string[] k_CmToolbarItems = { FreelookRigSelection.id };
+
+        public override VisualElement CreatePanelContent() => CreateContent(Layout.HorizontalToolbar);
+
+        /// <summary>Set this with your custom tools' IDs.</summary>
+        public static string[] customToolbarItems = null;
+        public IEnumerable<string> toolbarElements
+        {
+            get
+            {
+                if (customToolbarItems != null)
+                {
+                    var toolbarItems = new List<string>(k_CmToolbarItems);
+                    toolbarItems.AddRange(customToolbarItems);
+                    return toolbarItems;
+                }
+                
+                return k_CmToolbarItems;
+            }
+        }
+    }
+    
+    [EditorToolbarElement(id, typeof(SceneView))]
+    class FreelookRigSelection : EditorToolbarDropdown
+    {
+        public const string id = "FreelookRigSelection/Dropdown";
+        public static int SelectedRig;
+
+        public FreelookRigSelection()
+        {
+            tooltip = "Freelook Rig Selection";
+            clicked += FreelookRigSelectionMenu;
+            EditorApplication.update += ShadowSelectedRigName;
+            EditorApplication.update += DisplayIfRequired;
+        }
+
+        ~FreelookRigSelection()
+        {
+            clicked -= FreelookRigSelectionMenu;
+            EditorApplication.update -= ShadowSelectedRigName;
+            EditorApplication.update -= DisplayIfRequired;
+        }
+
+        Type m_FreelookRigSelectionType = typeof(FreelookRigSelection);
+        void DisplayIfRequired() => style.display = 
+            CinemachineSceneToolUtility.IsToolRequired(m_FreelookRigSelectionType) 
+                ? DisplayStyle.Flex : DisplayStyle.None;
+
+        // text is currently only visibly in Panel mode due to this bug: https://jira.unity3d.com/browse/STO-2278
+        void ShadowSelectedRigName() => text = CinemachineFreeLookEditor.RigNames[Mathf.Clamp(
+                SelectedRig, 0, CinemachineFreeLookEditor.RigNames.Length - 1)].text;
```

Comment:
The new rig icons look great, why don't you use them here too?

![image](https://user-images.githubusercontent.com/6424658/144324579-238a2575-beaf-4b0c-9827-7cd07da2f8ee.png)

Reply:
Good catch :)